### DISCUSSION TITLE
BLE Joystick on model load

### DISCUSSION THREAD

**Discussion by bill-healey** [ExpressLRS/ExpressLRS#2927] [Q&A] [answered]
Is there a way to start the BLE Joystick upon model load rather than having to navigate through the ELRS LUA script each time?

I tried pushing the CSRF command to turn it on via a LUA function, but no luck.

crossfireTelemetryPush(0x2D, { 0xEE, 0xEA, 16, 1 })

**Comment by VaSa4oK** [comment, marked as answer]
local elrs = assert(loadfile("/SCRIPTS/TOOLS/elrsV3.lua"))()
local function init_func()    
    elrs.init()
end
local function run_func()
 crossfireTelemetryPush(0x2D, { 0xEE, 0xEA, 29, 1 })  
    return 0
end
return { run=run_func, init=init_func }

**Reply by bill-healey** [reply]
Marking this as the answer since the code works assuming you substitute the correct command for your hardware (not always 29).  See the other comments for how to find it.

**Comment by VaSa4oK** [comment]
Try my script 😁
You just need to run the script, assign it to any button or switch. Through Spec. Function. Script may exist /SCRIPTS/FUNCTIONS/xxx.lua

**Reply by bill-healey** [reply]
I tried but couldn't get it to work.  How did you come up with command 29 rather than 16?

**Reply by VaSa4oK** [reply]
Maybe you have another ID 0xEE, 0xEA. What device and module do you have?

**Comment by pkendall64** [comment]
The values are not fixed. They are dynamically generated based on positions in a global array and will depend on the firmware version and features available on the TX module.

**Reply by VaSa4oK** [reply]
Yes, that's understandable. But I don't know how to write Lua scripts and how it should interact with the firmware.
If there are people who know, please fix the script, or make it right.

**Reply by bill-healey** [reply]
Got this working.  @VaSa4oK is right, the script works but you need the right ID, for me it was 19 instead of 29.

You can find it by making this small change to elrsV3.lua.  Change the fieldCommandDisplay function to this and it'll show the fieldID next to the name for BluetoothJoystick.  

```
local function fieldCommandDisplay(field, y, attr)
    lcd.drawText(10, y, "[" .. (field.id and field.name .. field.id or field.name) .. "]", attr + BOLD)
end
```

Run the elrsV3 script, look at the ID next to BLE Joystick, put that ID into the telemetry script, start the telemetry screen (or set it to autolaunch) if you're using the latest edgeTX and you're good to go.

**Comment by Oppa-YA** [comment]
> Is there a way to start the BLE Joystick upon model load rather than having to navigate through the ELRS LUA script each time?
> 
> I tried pushing the CSRF command to turn it on via a LUA function, but no luck.
> 
> crossfireTelemetryPush(0x2D, { 0xEE, 0xEA, 16, 1 })

Have you found the solution yet? I need the same thing.

**Reply by VaSa4oK** [reply]
It works great for me on TX16s.

**Reply by Oppa-YA** [reply]
I tried but sadly don't work, I think I need to find those numbers, (0x2D, { 0xEE, 0xEA, 16, 1 })     (get ID of my model)

**Reply by VaSa4oK** [reply]
16 change to 29

**Reply by VaSa4oK** [reply]
local elrs = assert(loadfile("/SCRIPTS/TOOLS/elrsV3.lua"))()
local function init_func()
elrs.init()
end
local function run_func()
crossfireTelemetryPush(0x2D, { 0xEE, 0xEA, 29, 1 })
return 0
end
return { run=run_func, init=init_func }

**Reply by Oppa-YA** [reply]
> local elrs = assert(loadfile("/SCRIPTS/TOOLS/elrsV3.lua"))() local function init_func() elrs.init() end local function run_func() crossfireTelemetryPush(0x2D, { 0xEE, 0xEA, 29, 1 }) return 0 end return { run=run_func, init=init_func }

I tried this but still does not work on Jumper T14, I think it need different ID, not sure how to get it. I am on latest Edge TX 2.10.4 and ELRS 3.5.0 firmware versions.

**Reply by VaSa4oK** [reply]
Sorry, I don't know anything about T14. I'm not a developer and I don't understand this. I made it for myself because I have TX16.

**Reply by VaSa4oK** [reply]
I tried different combinations. 16 and 1, changed from 1 to 29, and 1-6, respectively. I had Wi-Fi and a Wi-Fi backpack turned on. As soon as he got to BLE, he left it like that.

**Reply by Oppa-YA** [reply]
I see, thank you! I will search more, maybe will find it.
